builtin/logical/ad/path_config.go

```diff
@@ -0,0 +1,219 @@
+package ad
+
+import (
+	"context"
+
+	"github.com/fatih/structs"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/helper/activedirectory"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+const (
+	// This length is arbitrarily chosen but should work for
+	// most Active Directory minimum and maximum length settings.
+	// A bit of tongue-in-cheek since programmers love their base-2 exponents.
+	defaultPasswordLength = 64
+
+	// The number of minutes in 32 days.
+	defaultPasswordTTLs = 24 * 60 * 32
+)
+
+type configHandler struct {
+	logger hclog.Logger
+}
+
+func (h *configHandler) Path() *framework.Path {
+	return &framework.Path{
+		Pattern: "config",
+		Fields: map[string]*framework.FieldSchema{
+			"username": {
+				Type:        framework.TypeString,
+				Default:     "",
+				Description: "Username with sufficient permissions in Active Directory to administer passwords.",
+			},
+
+			"password": {
+				Type:        framework.TypeString,
+				Default:     "",
+				Description: "Password for username with sufficient permissions in Active Directory to administer passwords.",
+			},
+
+			"url": {
+				Type:        framework.TypeString,
+				Default:     "ldap://127.0.0.1",
+				Description: "LDAP URL to connect to (default: ldap://127.0.0.1). Multiple URLs can be specified by concatenating them with commas; they will be tried in-order.",
+			},
+
+			"certificate": {
+				Type:        framework.TypeString,
+				Default:     "",
+				Description: "CA certificate to use when verifying LDAP server certificate, must be x509 PEM encoded (optional)",
```

Comment:
Can we paraphrase the second sentence. There is a little bit of confusion as to whether the field is optional or PEM encoding is optional.

Also, do we want to have this accepted as a base64 encoded string to avoid any inconsistencies?

Reply:
Sure! All of these descriptions are from here https://github.com/hashicorp/vault/blob/master/builtin/credential/ldap/path_config.go#L84 and I will assume it's cool to not use the exact same phrasing throughout Vault. Same with the periods and such, I'll be consistent in whether I use them or not. 

I'm not sure what you mean with the latter question. If it's helpful, here's an example of a string that would be valid to pass in: https://github.com/hashicorp/vault/pull/4260/files#diff-6d58521f0386ed7fb2cec2533e2dbaddR166.